### ISSUE TITLE
False positive on sorted_first_last with first(where:)

### ISSUE THREAD

**Issue by joeydong** [realm/SwiftLint#3023]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

The `sorted_first_last` shouldn't trigger if there is a predicate passed to the `first` call

##### Complete output when running SwiftLint, including the stack trace and command used

**`MyPlayground.swift`**
```swift
import Foundation

struct RaceResult {
    let name: String
    let finish: Date
    let disqualified: Bool
}

let logbook = [
    RaceResult(name: "Troy", finish: Date.init(timeIntervalSince1970: 300), disqualified: false),
    RaceResult(name: "Rachel", finish: Date.init(timeIntervalSince1970: 100), disqualified: true),
    RaceResult(name: "Jessie", finish: Date.init(timeIntervalSince1970: 200), disqualified: false),
    RaceResult(name: "Tiffany", finish: Date.init(timeIntervalSince1970: 400), disqualified: false)
]

let winner = logbook
    .sorted { lhs, rhs -> Bool in
        lhs.finish < rhs.finish
    }
    .first { result -> Bool in
        !result.disqualified
    }

print(winner!.name) // Jessie
```

```bash
$ swiftlint lint --config .swiftlint.yml
Loading configuration from '.swiftlint.yml'
Linting Swift files at paths
Linting 'Contents.swift' (1/1)
/Users/joeydong/Downloads/MyPlayground.playground/Contents.swift:16:14: warning: Min or Max over Sorted First or Last Violation: Prefer using `min()` or `max()` over `sorted().first` or `sorted().last` (sorted_first_last)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.38.0
* Installation method used (Homebrew, CocoaPods, building from source, etc)? Homebrew
* Paste your configuration file:
```yml
opt_in_rules:
  - sorted_first_last
```
* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? No
* Which Xcode version are you using (check `xcodebuild -version`)?
```
Xcode 11.3
Build version 11C29
```

**Comment by joeydong** [comment]
now that i really think about it, i suppose you can also do `filter` then `max`:

```
import Foundation

struct RaceResult {
    let name: String
    let finish: Date
    let disqualified: Bool
}

let logbook = [
    RaceResult(name: "Troy", finish: Date.init(timeIntervalSince1970: 300), disqualified: false),
    RaceResult(name: "Rachel", finish: Date.init(timeIntervalSince1970: 100), disqualified: true),
    RaceResult(name: "Jessie", finish: Date.init(timeIntervalSince1970: 200), disqualified: false),
    RaceResult(name: "Tiffany", finish: Date.init(timeIntervalSince1970: 400), disqualified: false)
]

let winner = logbook
    .filter { result -> Bool in
        !result.disqualified
    }
    .max { lhs, rhs -> Bool in
        lhs.finish > rhs.finish
    }

print(winner!.name) // Jessie
```

**Comment by jpsim** [comment]
If your filter/first predicate is more expensive than sorting, it might be best to keep the sorted+first version. I'm labeling this as a bug.
